src/Providers/Http/DTO/Request.php

```diff
@@ -0,0 +1,371 @@
+            $new->data = null;
+        } elseif (is_array($data)) {
+            $new->data = $data;
+            $new->body = null;
+        } else {
+            $new->data = null;
+            $new->body = null;
+        }
+        return $new;
+    }
+
+    /**
+     * Gets the request data array.
+     *
+     * @since n.e.x.t
+     *
+     * @return array<string, mixed>|null The request data array.
+     */
+    public function getData(): ?array
+    {
+        return $this->data;
+    }
+
+    /**
+     * Gets the request body string.
+     *
+     * @since n.e.x.t
+     *
+     * @return string|null The request body string.
+     */
+    public function getBodyString(): ?string
+    {
+        return $this->body;
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @since n.e.x.t
+     */
+    public static function getJsonSchema(): array
+    {
+        return [
+            'type' => 'object',
+            'properties' => [
+                self::KEY_METHOD => [
+                    'type' => 'string',
+                    'description' => 'The HTTP method.',
+                ],
+                self::KEY_URI => [
+                    'type' => 'string',
+                    'description' => 'The request URI.',
+                ],
+                self::KEY_HEADERS => [
+                    'type' => 'object',
+                    'additionalProperties' => [
+                        'type' => 'array',
+                        'items' => ['type' => 'string'],
+                    ],
+                    'description' => 'The request headers.',
+                ],
+                self::KEY_DATA => [
+                    'type' => ['string', 'array', 'null'],
+                    'description' => 'The request data.',
+                ],
+            ],
+            'required' => [self::KEY_METHOD, self::KEY_URI, self::KEY_HEADERS],
+        ];
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @since n.e.x.t
+     *
+     * @return RequestArrayShape
+     */
+    public function toArray(): array
+    {
+        $array = [
+            self::KEY_METHOD => $this->method->value,
+            self::KEY_URI => $this->uri,
+            self::KEY_HEADERS => $this->headers->getAll(),
+        ];
+
+        // Include whichever data type is set
+        if ($this->body !== null) {
+            $array[self::KEY_DATA] = $this->body;
+        } elseif ($this->data !== null) {
+            $array[self::KEY_DATA] = $this->data;
+        }
+
+        return $array;
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @since n.e.x.t
+     */
+    public static function fromArray(array $array): self
+    {
+        static::validateFromArrayData($array, [self::KEY_METHOD, self::KEY_URI, self::KEY_HEADERS]);
+
+        return new self(
+            HttpMethodEnum::from($array[self::KEY_METHOD]),
+            $array[self::KEY_URI],
+            $array[self::KEY_HEADERS] ?? [],
+            $array[self::KEY_DATA] ?? null
```

Comment:
Per the above suggestions, this would become:
```suggestion
            $array[self::KEY_BODY] ?? null
```

Reply:
Resolved in adfcd5f